File: Java_Programmers/1단계/src/하샤드수.java
public class 하샤드수 {
    public boolean solution(int x) {
        String[] str = String.valueOf(x).split("");
        int sum = 0;
        for (String s : str) {
            sum += Integer.parseInt(s);
        }
        return x % sum == 0;
    }
    public boolean solution2(int num){
        int sum = 0;
        int n = num;
        while(n>0){
            sum += n%10;
            n /= 10;
        }
        return num % sum == 0;
    }
    public static void main(String[] args) {
        하샤드수 sol = new 하샤드수();
        int x = 12;
        System.out.println(sol.solution(x));
        System.out.println(sol.solution2(x));
    }
}

File: python/python_Programmers/1단계/하샤드수.py
def solution(x):
    hap = 0
    x = str(x)
    for i in range(len(x)):
        hap += int(x[i])

    if int(x) % hap == 0:
        return True
    return False

# 모범 풀이
def solution2(n):
    # n은 하샤드 수 인가요?
    return n % sum([int(c) for c in str(n)]) == 0


arr = 13
print(solution(arr))
print(solution2(arr))